src/hash/zelbet_gadget/bar.rs

```diff
@@ -18,27 +18,24 @@ use dusk_plonk::prelude::*;
 fn witness_composition_gadget(
     composer: &mut TurboComposer,
     witness_decomposed: &[Witness; 27],
-    zelbet_witnesses: &ZelbetWitnesses,
 ) -> Witness {
     // Add relevant gates for the decomposition
     // Start: acc = x_1·s_2 + x_2
     let constraint = Constraint::new()
-        .mult(BLS_ONE)
-        .a(witness_decomposed[26])
-        .b(zelbet_witnesses.s_i_decomposition[25])
-        .fourth(BLS_ONE)
-        .d(witness_decomposed[25]);
-    let mut acc = composer.gate_mul(constraint);
-
-    // acc = (((( x_1·s_2 + x_2 )·s_3 + x_3 )·s_4 + x_4) ... )·s_27 + x_27
-    for k in 1..26 {
+        .left(BlsScalar::from(S_I[1] as u64))
+        .a(witness_decomposed[0])
+        .right(BLS_ONE)
```

Comment:
```suggestion
        .right(1)
```

Reply:
And if @moCello  want to avoid the conversion all the 25 times, can pass directly `BlsScalar::one()` as stated in the other comment, since it's a `const fn`

```suggestion
        .right(BlsScalar::one())
```